CesiumIonClient/src/Connection.cpp

```diff
@@ -364,6 +395,60 @@ CesiumAsync::Future<Response<Profile>> Connection::me() const {
           });
 }
 
+/* static */ CesiumAsync::Future<Response<ApplicationData>>
+CesiumIonClient::Connection::appData(
+    const CesiumAsync::AsyncSystem& asyncSystem,
+    const std::shared_ptr<CesiumAsync::IAssetAccessor>& pAssetAccessor,
+    const std::string& apiUrl) {
+  return pAssetAccessor
+      ->get(
+          asyncSystem,
+          CesiumUtility::Uri::resolve(apiUrl, "/appData"),
+          {{"Accept", "application/json"}})
+      .thenInMainThread([](std::shared_ptr<CesiumAsync::IAssetRequest>&&
```

Comment:
Why in the main thread? I totally accept the answer "because every other function in this file resolves the future in the main thread" and I would have written it the same way for that reason, but it would be nice to have a comment somewhere explaining why this has to be done in the main thread.

Reply:
I'm not seeing anything in this lambda that's obviously thread-unsafe, or can even be accessed by other functions. It's just building a result for the caller.

I would choose `::thenImmediately`. Odds are this isn't a performance critical path though, difference is probably academic.